chpl/chpl-auth/src/main/java/gov/healthit/chpl/auth/dao/impl/UserResetTokenDAOImpl.java

```diff
@@ -0,0 +1,89 @@
+package gov.healthit.chpl.auth.dao.impl;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import javax.persistence.Query;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.stereotype.Repository;
+import gov.healthit.chpl.auth.BaseDAOImpl;
+import gov.healthit.chpl.auth.Util;
+import gov.healthit.chpl.auth.dao.UserResetTokenDAO;
+import gov.healthit.chpl.auth.dto.UserResetTokenDTO;
+import gov.healthit.chpl.auth.entity.UserResetTokenEntity;
+
+@Repository(value="userResetTokenDAO")
+public class UserResetTokenDAOImpl extends BaseDAOImpl implements UserResetTokenDAO {
+    private static final Logger logger = LogManager.getLogger(UserResetTokenDAOImpl.class);
+
+    public UserResetTokenDTO create(String resetToken, Long userId) {
+        UserResetTokenEntity entity = new UserResetTokenEntity();
+        entity.setUserResetToken(resetToken);
+        entity.setUserId(userId);
+        entity.setCreationDate(new Date());
+        entity.setLastModifiedDate(new Date());
```

Comment:
Creation date and last modified date should not need to be set here since the database sets them. You may need to mark them as insertable = false, updatable = false in the entity.